CMakeLists.txt

```diff
@@ -272,3 +272,12 @@ if (BENCHMARK_ENABLE_TESTING)
   endif()
   add_subdirectory(test)
 endif()
+
+set(CPACK_GENERATOR "DEB")
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Google Inc.")
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Google Benchmark library.")
+set(CPACK_PACKAGE_DESCRIPTION "A library to support the benchmarking of functions, similar to unit-tests.")
+set(CPACK_DEBIAN_PACKAGE_SECTION "Development")
+
+include(CPack)
```

Comment:
is this the cmake 3.0 version of cpack? we require 2.8.12 and the cpack documentation for that looks different: https://cmake.org/cmake/help/v2.8.12/cpack.html

Reply:
I honestly didn't look into that.

I've done a quick search and the only variable I can't find in the documentation is `CPACK_PACKAGE_DESCRIPTION`, which looks like it should be either removed or replaced with `CPACK_PACKAGE_DESCRIPTION_FILE` + a text file withe the long description.

In this file I see `cmake_minimum_required (VERSION 2.8.12)`, and my version of `cmake` is 3.5.1.
Is there any way on your end you could tried with 2.8.12?
Otherwise, I could try building https://gitlab.kitware.com/cmake/cmake/tree/v2.8.12.2 and giving it a try.

Reply:
I'd use CPACK_DESCRIPTION_SUMMARY.

We could also update cmake to 3.5.1: #801

Reply:
@dominichamon Great to hear you're bumping up to cmake 3!

I already use both `CPACK_DESCRIPTION_SUMMARY` and `CPACK_PACKAGE_DESCRIPTION`, for the short and long description.

This is how things look like in `aptitude`:
![benchmark_aptitude](https://user-images.githubusercontent.com/382167/56227849-adcfe080-6076-11e9-9f24-a2cf0f6209e2.png)

It looks indeed that the `CPACK_PACKAGE_DESCRIPTION` isn't showing up. :thinking: 

Same with:
``` bash
$ apt-cache show benchmark
Package: benchmark
Version: 1.4.0
Section: Development
Priority: optional
Architecture: amd64
Installed-Size: 916
Maintainer: Google Inc.
Description: Google Benchmark library.
Description-md5: 3f3fbd5479e6d7d7b1ab982eeb7ee786
Filename: pool/main/benchmark-1.4.0-Linux.deb
SHA1: 0b2e4c623651fbd1480e9df15090c9b7f18c35a6
SHA256: 12fa22510f985e01b8a9ca238937ef771762322f59fa091e28b92fb2c3200abd
Size: 228010
```

Not a big deal, since it's optional. Let me know if you want me to remove it for now, since it's doing nothing, or explore how to fix it.

Reply:
(https://packages.ubuntu.com/search?keywords=benchmark&searchon=names says it's packaged already)

Reply:
Good call. I think I missed that b/c I'm on Ubuntu 16.04.

I think this contribution is still useful, but I could certainly consider taking the debian from 18.04 or 18.10. It seems that they packaged benchmark 1.3.0 though, so this contribution at least makes it easier to generate a debian locally for the latest version.

Anyway, it's up to you if you want this close, or if you want me to do something else before merging. :smiley:

Reply:
I don't have any strong opinion on this, seems not too intrusive,
although not a full-fledged package. I'll leave this up to @dominichamon.